models/ssh_key.go

```diff
@@ -717,8 +742,11 @@ func RegeneratePublicKeys(t io.StringWriter) error {
 
 func regeneratePublicKeys(e Engine, t io.StringWriter) error {
 	err := e.Iterate(new(PublicKey), func(idx int, bean interface{}) (err error) {
-		_, err = t.WriteString((bean.(*PublicKey)).AuthorizedString())
-		return err
+		if bean.(*PublicKey).Type != KeyTypePrincipal {
+			_, err = t.WriteString((bean.(*PublicKey)).AuthorizedString())
+			return err
+		}
+		return nil
 	})
 	if err != nil {
 		return err
```

Comment:
```suggestion
	if err := e.Where("type != ?", KeyTypePrincipal).Iterate(new(PublicKey), func(idx int, bean interface{}) (err error) {
		_, err = t.WriteString((bean.(*PublicKey)).AuthorizedString())
		return err
	}); err != nil {
		return err
```

Reply:
Can't seem to commit it with github. "applying suggestions on deleted lines is not supported", but I'll add the change manually